tests/Rules/Exceptions/CatchWithUnthrownExceptionRuleTest.php

```diff
@@ -0,0 +1,24 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Rules\Exceptions;
+
+use PHPStan\Rules\Rule;
+use PHPStan\Testing\RuleTestCase;
+
+/**
+ * @extends RuleTestCase<CatchWithUnthrownExceptionRule>
+ */
+class CatchWithUnthrownExceptionRuleTest extends RuleTestCase
+{
+
+	protected function getRule(): Rule
+	{
+		return new CatchWithUnthrownExceptionRule();
```

Comment:
You should be able to get rid of the BC message with: `self::getContainer()->getByType(CatchWithUnthrownExceptionRule::class);`